src/tools/compiletest/src/common.rs

```diff
@@ -456,11 +457,12 @@ pub enum Endian {
 }
 
 impl TargetCfg {
-    fn new(rustc_path: &Path, target: &str) -> TargetCfg {
+    fn new(rustc_path: &Path, target: &str, target_rustcflags: &Option<String>) -> TargetCfg {
         let output = match Command::new(rustc_path)
             .arg("--print=cfg")
             .arg("--target")
             .arg(target)
+            .args(target_rustcflags.into_iter().map(|s| s.split_whitespace()).flatten())
```

Comment:
Could you refactor the target_rustcflags definition to be a `Option<Vec<String>>`? The semantic separation of separate flags wouldn't be lost that way.

Reply:
To confirm, I need to refactor all usages of `target_rustcflags`? Or just refactor in this function? 

Wanting to confirm, as refactoring all usages brings it out of sync with `host_rustcflags`, which may confuse future users of compiletest

Reply:
I think if my glance at the arg parsing was right we currently take multiple arguments (i.e. --target-rustcflags can be specified more than once) and squash them into a single string by joining with `" "`. My thinking was to avoid doing the squash and then split here.

I agree that it makes sense to do the same for host-rustcflags if we do that.

Are we splitting on space already elsewhere in compiletest? I think if so I'm happy to r+ this and leave that improvement to a follow-up.

Reply:
Correct -- `compiletest` uses splitting on space throughout for `host_rustcflags` and `target_rustcflags` on use by https://github.com/rust-lang/rust/blob/master/src/tools/compiletest/src/runtest.rs#L2128

I agree refactoring will be good, though may be good as a follow-up!